deepinv/datasets/datagenerator.py

```diff
@@ -1,76 +1,125 @@
+from __future__ import annotations
+from typing import TYPE_CHECKING, Union, Callable, Tuple
+
 from tqdm import tqdm
 import os
 import h5py
 import torch
-from torch.utils.data import DataLoader, Subset
+from torch import Tensor
+from torch.utils.data import DataLoader, Subset, Dataset
 from torch.utils import data
 
+if TYPE_CHECKING:
+    from deepinv.physics import Physics
+    from deepinv.physics.generator import PhysicsGenerator
+    from deepinv.transform import Transform
+
 
 class HDF5Dataset(data.Dataset):
     r"""
     DeepInverse HDF5 dataset with signal/measurement pairs.
 
+    If there is no training ground truth (i.e. ``x_train``) in the dataset file,
+    the dataset returns the measurement again as the signal.
+
+    Optionally also return physics generator params as a dict per sample, if one was used during data generation.
+
     :param str path: Path to the folder containing the dataset (one or multiple HDF5 files).
     :param bool train: Set to ``True`` for training and ``False`` for testing.
-    :param transform: A torchvision transform to apply to the data.
+    :param Transform, Callable transform: A deepinv or torchvision transform to apply to the data.
+    :param bool load_physics_generator_params: load physics generator params from dataset if they exist (e.g. if dataset created with :meth:`deepinv.datasets.generate_dataset`)
+    :param torch.dtype, str dtype: cast all real-valued data to this dtype.
+    :param torch.dtype, str complex_dtype: cast all complex-valued data to this dtype.
```

Comment:
Maybe we can be a bit more precise on the behaviour of such function: e.g. would imaginary part be ignored?